include/tenstorrent/spi_flash_buf.h

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) 2025 Tenstorrent AI ULC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _SPI_FLASH_BUF_H_
+#define _SPI_FLASH_BUF_H_
+
+#include <stdint.h>
+#include <stddef.h>
+
+#include <zephyr/device.h>
+
+#define TEMP_SPI_BUFFER_SIZE 4096
+
+int spi_arc_dma_transfer_to_tile(const struct device *dev, const char *tag, uint8_t *buf,
+				 size_t buf_size, uint8_t *tlb_dst);
```

Comment:
It should be possible to perform a `tt_boot_fs_find_fd_by_tag()` once, outside of this function, and then to pass the "file" offset and size in spi flash, so I would suggest minor changes to the signature of this function.

This will help performance-wise especially in the case where the same firmware image is loaded over and over to multiple e.g. memory controllers.  

```suggestion
int spi_arc_dma_transfer_to_tile(const struct device *dev, size_t spi_image_offset, size_t spi_image_size, uint8_t *buf, size_t buf_size, uint8_t *tlb_dst);
```